docs/src/content/docs/tutorials/url-shortener.md

```diff
@@ -0,0 +1,176 @@
+
+        // Serialize URL struct to JSON and store it in DiceDB
+        urlData, err := json.Marshal(requestBody)
+        if err != nil {
+            c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to save URL"})
+            return
+        }
+
+        if err := db.Set(context.Background(), shortID, urlData, 0).Err(); err != nil {
+            c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to save URL"})
+            return
+        }
+
+        c.JSON(http.StatusCreated, gin.H{"short_url": requestBody.ShortURL})
+    }
+
+    // Redirects to the original URL based on the short URL ID
+    func RedirectURL(c *gin.Context) {
+        id := c.Param("id")
+
+        // Retrieve stored URL data from DiceDB
+        urlData, err := db.Get(context.Background(), id).Result()
+        if err != nil {
+            c.JSON(http.StatusNotFound, gin.H{"error": "URL not found"})
+            return
+        }
+
+        // Deserialize JSON data back into URL struct
+        var url URL
+        if err := json.Unmarshal([]byte(urlData), &url); err != nil {
+            c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to decode URL data"})
+            return
+        }
+
+        // Redirect user to the original long URL
+        c.Redirect(http.StatusFound, url.LongURL)
+    }
+
+    func main() {
+        router := gin.Default()
+
+        // Define endpoints for creating short URLs and redirecting
+        router.POST("/shorten", CreateShortURL)
+        router.GET("/:id", RedirectURL)
+
+        // Start the server on port 8080
+        if err := router.Run(":8080"); err != nil {
+            log.Fatal("Failed to start server:", err)
+        }
+    }
+    ```
+
+## Explanation of Key Parts
+1. Database Initialization: The `init()` function sets up a DiceDB client to connect to `localhost:7379`.
+2. `CreateShortURL` Endpoint: Handles the `/shorten` route. It generates a unique ID, constructs the short URL, serializes the URL data, and saves it in DiceDB.
+3. `RedirectURL` Endpoint: Handles the `/:id` route. It retrieves the original URL by the short ID from DiceDB and redirects the user to it.
+4. Starting the Server: The `main` function starts the Gin server on port `8080`.
+
+## Starting the application server
+
+1. Start the application
+   ```bash
+   go run main.go
+   ```
+   This will start the application server on port 8080 by default, you should see output similar to
+   ```bash
+   [GIN-debug] Listening and serving HTTP on :8080
+   ```
+
+## Interacting with the application
+
+1. Start DiceDB: Ensure DiceDB is running.
+2. Test the API:
+    - Shorten URL:
+        Send a POST request to `/shorten` with JSON body on Postman:
+        ```
+        {
+            "long_url": "https://example.com"
+        }
+        ```
+
+        OR
+
+        ```curl
+        curl -X POST -H "Content-Type: application/json" -d '{"long_url": "https://example.com"}' http://localhost:8080/shorten
+        ```
```

Comment:
```suggestion
**Using `curl`:**

    ```bash
curl -X POST -H "Content-Type: application/json" -d '{"long_url": "https://example.com"}' http://localhost:8080/shorten
    ```

**Response:**

    ```json
{
  "short_url": "http://localhost:8080/<short ID generated by the server>"
}
    ```
```